packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -28,31 +32,57 @@ export const NotificationsContextProvider = ({
   children,
   unreadCount = 0,
 }: NotificationsContextProviderProps): ReactElement => {
+  const isExtension = !!process.env.TARGET_BROWSER;
   const { user } = useContext(AuthContext);
-  const [hasPermission, setHasPermission] = useState(
-    globalThis.window?.Notification?.permission === 'granted',
-  );
-
-  const notificationsAvailable = () => {
-    return !!globalThis.window?.Notification;
-  };
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [hasPermission, setHasPermission] = useState(false);
 
-  const requestPermission = async (): Promise<NotificationPermission> => {
+  const onTogglePermission = async (): Promise<NotificationPermission> => {
     if (!user) return 'default';
 
+    if (hasPermission) {
+      await OneSignalReact.setSubscription(false);
+      setHasPermission(false);
+      return 'denied';
+    }
+
     const result = await globalThis.window?.Notification?.requestPermission();
-    setHasPermission(result === 'granted');
+    const isGranted = result === 'granted';
+    setHasPermission(isGranted);
+    OneSignalReact.setSubscription(isGranted);
+
     return result;
   };
 
-  const data = useMemo(() => {
-    return {
+  useEffect(() => {
+    if (isInitialized || !user || isExtension) {
```

Comment:
Another point, before merging this PR we have to decide what to do with the extension. Let's discuss the possibilities. We can have a meeting about it or you can suggest

Reply:
Surely. I am planning to finish off the webapp first, so I applied some conditions to ensure the extension won't be affected in any way after webapp implementation. Once done, I will go back to the extension and see what can work.

Reply:
So I'll keep it as a blocker. We can't merge it before the extension support

Reply:
Cool. Would that be a blocker for this PR only or as a whole in the notifications intermediate branch?

Reply:
For this PR